### PR TITLE
Simple backend-independent counterexamples

### PR DESCRIPTION
**Pull request by viper-admin**
>  **Pull request** :twisted_rightwards_arrows: created by **@marcoeilers** on 2019-08-05 11:50
> Last updated on 2019-11-05 12:25
> Original Bitbucket pull request id: 63
>
> Participants:
>
> * **@mschwerhoff** (reviewer)
>
> Source: https://github.com/viperproject/silicon/commit/9e95b3d70fdea98fb8ed060b8edf29bc03cda6a4 on branch `meilers_counterexamples`
> Destination: https://github.com/viperproject/silicon/commit/76f21ec30b0fe9e7be00e4e6f56db9fd42984a42 on branch `master`
>
> State: **`OPEN`**

Silicon part of Silver PR [https://github.com/viperproject/silver/pull/431](https://github.com/viperproject/silver/pull/431)

### CODE REVIEW

File: src/main/scala/decider/Z3ProverStdIO.scala [hunk 1]
```diff
@@ -235,10 +236,19 @@ class Z3ProverStdIO(uniqueId: String,
   }
 
   private def getModel(): Unit = {
-    if (Verifier.config.ideModeAdvanced()) {
+    if (Verifier.config.ideModeAdvanced() || Verifier.config.counterexample.toOption.isDefined) {
       writeLine("(get-model)")
-      val model = readModel().trim()
-      println(model + "\r\n")
+
+      if (Verifier.config.ideModeAdvanced()){
```
Comment (mschwerhoff) [inline src/main/scala/decider/Z3ProverStdIO.scala:242]: @marcoeilers Any idea if the if-branch is still needed? A `println` of a model seems not really helpful, in particular in the context of something apparently related to the IDE (I don't know/remember who contributed that code).

Comment (marcoeilers) [reply src/main/scala/decider/Z3ProverStdIO.scala:242]: I don't know of anything that would use it. 
@aterga @ArquintL Do any of your student projects or your MSc thesis use the counterexample printed to the command line?

Comment (ArquintL) [reply src/main/scala/decider/Z3ProverStdIO.scala:242]: My MSc thesis does not use anything printed to standard out. I've just double-checked the PR of my MSc thesis and original if branch was already there back then

Comment (marcoeilers) [reply src/main/scala/decider/Z3ProverStdIO.scala:242]: I'll just remove it then and we'll see if anyone complains.

**Pull request merged**